### PR TITLE
Add citation

### PR DESCRIPTION
**Pull request by ziofil**
- Adds citations for symplectic and orthogonal optimization
- fixes variable/function names to match math

### CODE REVIEW

File: mrmustard/utils/training.py [hunk 1]
```diff
@@ -146,8 +146,15 @@ def new_orthogonal(num_modes: int) -> Tensor:
 def update_symplectic(
     symplectic_params: Sequence[Trainable], symplectic_grads: Sequence[Tensor], symplectic_lr: float
 ):
-    for S, dS_riemann in zip(symplectic_params, symplectic_grads):
-        Y = math.riemann_to_symplectic(S, dS_riemann)
+    r"""
+    Updates the symplectic parameters using the given symplectic gradients.
+    Implemented from:
```
Comment (thisac) [inline mrmustard/utils/training.py:151]:
```suggestion
    r"""Updates the symplectic parameters using the given symplectic gradients.

    Implemented from:
```

File: mrmustard/utils/training.py [hunk 2]
```diff
@@ -158,9 +165,20 @@ def update_symplectic(
 def update_orthogonal(
     orthogonal_params: Sequence[Trainable], orthogonal_grads: Sequence[Tensor], orthogonal_lr: float
 ):
-    for O, dO_riemann in zip(orthogonal_params, orthogonal_grads):
-        D = 0.5 * (dO_riemann - math.matmul(math.matmul(O, math.transpose(dO_riemann)), O))
-        new_value = math.matmul(O, math.expm(orthogonal_lr * math.matmul(math.transpose(D), O)))
+    r"""
+    Updates the orthogonal parameters using the given orthogonal gradients.
+    Implemented from:
```
Comment (thisac) [inline mrmustard/utils/training.py:170]:
```suggestion
    r"""Updates the orthogonal parameters using the given orthogonal gradients.

    Implemented from:
```

**Review by thisac — APPROVED** [review]

**Comment by sduquemesa** [comment]
What about creating a bibtex file for citations?

File: mrmustard/math/math_interface.py [hunk 3]
```diff
@@ -1100,18 +1100,22 @@ def convolve_probs(self, prob: Tensor, other: Tensor) -> Tensor:
...
-        r"""Convert the Riemannian gradient to a symplectic gradient.
+    def euclidean_to_symplectic(self, S: Matrix, dS_euclidean: Matrix) -> Matrix:
+        r"""Convert the Euclidean gradient to a Riemannian gradient on the
+        tangent bundle of the symplectic manifold.
 
-        TODO: add citation (S.Fiori)
+        Implemented from:
+        Wang J, Sun H, Fiori S. A Riemannian‐steepest‐descent approach
+        for optimization on the real symplectic group.
+        Mathematical Methods in the Applied Sciences. 2018 Jul 30;41(11):4273-86.
```
Comment (sduquemesa) [inline mrmustard/math/math_interface.py:1110]:
```suggestion
        Implemented from:
            Wang J, Sun H, Fiori S. A Riemannian‐steepest‐descent approach
            for optimization on the real symplectic group.
            Mathematical Methods in the Applied Sciences. 2018 Jul 30;41(11):4273-86.
```

**Comment by ziofil** [comment]
I hit the button assuming the suggestions would get merged in - did that happen? 😬

**Comment by sduquemesa** [comment]
😮 you had to accept and commit them

**Comment by ziofil** [comment]
I'm making a new PR

**Pull request merged**